rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -33,14 +34,19 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         final JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
         if (requestCode == BARCODE_REQUEST_CODE && resultCode == RESULT_OK && data != null) {
             try {
-                String barcodeVals = getBarcodeValsAsCSV(data);
-                jsonApi.writeValue(widgetArgs.getStepName(),
-                        widgetArgs.getJsonObject().optString(JsonFormConstants.KEY),
-                        barcodeVals, "", "", "", false);
+                if (data.getBooleanExtra(CovidConstants.IntentKeys.BARCODE_SENSOR_TRIGGER, false)) {
+                    sensorTriggered();
+                }
```

Comment:
These logic should only be in the OneScanBarcodeFactory, see https://github.com/onaio/rdt-standard/pull/463/commits/8f45c0a0fb14c8796e0faa5de009df1aa5fe2b9b#r484807540.

Reply:
fixed.